Applications/FileIO/GocadIO/GocadAsciiReader.cpp

```diff
@@ -108,7 +160,21 @@ MeshLib::Mesh* GocadTSurfaceReader::readMesh(std::ifstream& in,
                 return nullptr;
             }
         }
-        else if (str[0] == "TFACE")
+        else if (type == GOCAD_DATA_TYPE::PLINE && str[0] == "ILINE")
```

Comment:
This is copy of the following if for TSURF. Please avoid code duplications.

Reply:
:heavy_check_mark:

Reply:
The duplication is indeed resolved, but now the mesh creation logic is mixed up with the parsing of the cases. It is difficult to add new cases there.
It's up to you, but I'd have written something like
```c++
auto parse_mesh = [&](auto node_element_parser) {
    std::vector<MeshLib::Node*> nodes;
    std::vector<MeshLib::Element*> elems;
    std::map<std::size_t, std::size_t> node_id_map;
    if (!node_element_parser(in, nodes, elems, node_id_map, mesh_prop))
    {
        ERR(...);
        return nullptr;
    }
    return std::make_unique<MeshLib::Mesh>(mesh_name, ...);
};
```
And in the conditions:
```c++
if (type == TFACE)
{
    return parse_mesh(parseSurface);
}
if (type == PLINE)
{
    return parse_mesh(parseLine);
}
```

Reply:
Honestly, I have no idea what happens there. And I know that it's difficult to add cases now, that's why it was written before such that cases were independent of each other.

Reply:
To be more specific, I'd very much prefer the code the way it was originally written. That'd be six lines of duplicate code, which isn't a big issue in my opinion if it means the code is easier to read.